app/controllers/api/v1/users_controller.rb

```diff
@@ -26,12 +26,22 @@ def create
         # TODO: amir - ensure accessibility for unauthenticated requests only.
         params[:user][:language] = I18n.default_locale if params[:user][:language].blank?
 
-        user = UserCreator.new(user_params:, provider: current_provider, role: default_role).call
+        registration_method = SettingGetter.new(setting_name: 'RegistrationMethod', provider: current_provider).call
+
+        return render_error errors: 'InviteInvalid' if registration_method == SiteSetting::REGISTRATION_METHODS[:invite] && !valid_invite_token
+
+        user = UserCreator.new(user_params: user_params.except(:invite_token), provider: current_provider, role: default_role).call
 
         # TODO: Add proper error logging for non-verified token hcaptcha
         return render_error errors: user.errors.to_a if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
 
         if user.save
+          # Delete invitation (ignore whether it exists or not)
+          if registration_method == SiteSetting::REGISTRATION_METHODS[:invite]
+            Invitation.delete_by(email: user_params[:email], provider: current_provider,
```

Comment:
This isn't a problem here, but it's recommended to avoid the use of `delete_by` unless required since it bypasses dependencies,